packages/test/test-end-to-end-tests/src/test/container.spec.ts

```diff
@@ -291,4 +293,104 @@ describeNoCompat("Container", (getTestObjectProvider) => {
         assert.strictEqual(container.closed, true);
         assert.strictEqual(pendingLocalState.url, (container.resolvedUrl as IFluidResolvedUrl).url);
     });
+
+    it("can call connect() and disconnect() on Container", async () => {
+        const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+            runtime.IFluidHandleContext.resolveHandle(request);
+        const runtimeFactory = (_?: unknown) => new TestContainerRuntimeFactory(
+            TestDataObjectType,
+            getDataStoreFactory(),
+            {},
+            [innerRequestHandler]);
+        const localTestObjectProvider = new TestObjectProvider(
+            Loader,
+            provider.driver,
+            runtimeFactory);
+
+        const container = await localTestObjectProvider.makeTestContainer() as Container;
+        await timeoutPromise(
+            (resolve) => container.once("connected", () => resolve()),
+            {durationMs: timeoutMs, errorMsg: "container initial connection timeout"},
+        );
+        assert.strictEqual(
+            container.connectionState, ConnectionState.Connected,
+            "container is not connected when loaded",
+        );
+
+        container.disconnect();
```

Comment:
Can you wait for the "disconnected" event here like you do for the "connected" event? It'd be nice to test that.

Reply:
Actually, as I mentioned in a comment earlier today, I realized `disconnect` is fully synchronous and the `disconnected` event should fire under this call. So maybe do something like this (and drop the `timeoutPromise`)?

```suggestion
        let disconnectedEventFired = false;
        container.once("disconnected", () => { disconnectedEventFired = true; });
        container.disconnect();
        assert(disconnectedEventFired, "disconnected event didn't fire when calling container.disconnect");
```